backend/memory/memory.go

```diff
@@ -303,7 +303,9 @@ func (f *Fs) list(ctx context.Context, bucket, directory, prefix string, addBuck
 					}
 					_, found := dirs[dir]
 					if !found {
+						b.mu.RUnlock()
```

Comment:
This is ranging over `b.objects` here - I don't think it is safe to release the lock as it gives the opportunity for other methods to grab it and mutate `b.objects`.